packages/search/addon/components/navi-search-result/asset.js

```diff
@@ -7,18 +7,18 @@
 
 import Component from '@glimmer/component';
 import { pluralize } from 'ember-inflector';
+import { set } from '@ember/object';
 
 export default class NaviAssetSearchResultComponent extends Component {
   /**
    * @property {Array} results
    */
   get results() {
     return this.args.data?.map(value => {
-      return {
-        route: this._getRouteFor(value),
-        type: value?.constructor?.modelName,
-        ...value
-      };
+      set(value, 'route', this._getRouteFor(value));
+      set(value, 'type', value.constructor?.modelName);
+      set(value, 'icon', value.type === 'report' ? 'file-text' : 'bar-chart');
```

Comment:
Can we not use Ember.set? 
```
value.route = ...
value.type = ...
```

Reply:
No, because `data` is tracked from `navi-search-bar`'s `searchResults` array, if I don't use set ember compaints.